### PR TITLE
[PVR] Refactoring: Remove GUI code from PVR core components, part 2.

### PR DESCRIPTION
**Pull request by ksooo**
This is a followup to #10870 and brings us a step closer to a PVR core components without gui dependencies. No functional changes here.

Additionally, a functional change/fix/improvement for "continue last channel on startup" is included in this PR.

@Jalle19 for review?

### CODE REVIEW

File: xbmc/pvr/PVRGUIActions.cpp [hunk 1]
```diff
@@ -778,21 +778,27 @@ namespace PVR
...
+    {
+      // Obtain previous 'was playing on last app quit' flag and reset it, then.
+      channel->SetWasPlayingOnLastQuit(false, bWasPlaying);
+    }
+
     int iPlayMode = CServiceBroker::GetSettings().GetInt(CSettings::SETTING_PVRPLAYBACK_STARTLAST);
     if (iPlayMode == CONTINUE_LAST_CHANNEL_OFF)
       return false;
 
-    const CFileItemPtr item(g_PVRChannelGroups->GetLastPlayedChannel());
```
Comment (Jalle19) [inline xbmc/pvr/PVRGUIActions.cpp]: Is `GetLastPlayedChannel` still used?

File: xbmc/pvr/PVRJobs.h [hunk 2]
```diff
@@ -39,4 +39,14 @@ namespace PVR
     bool m_bOnOff;
   };
 
+  class CPVRContinueLastChannelJob : public CJob
+  {
+  public:
+    CPVRContinueLastChannelJob() = default;
+    virtual ~CPVRContinueLastChannelJob() {}
```
Comment (Jalle19) [inline xbmc/pvr/PVRJobs.h]: `= default()`?

File: xbmc/pvr/PVRManager.h [hunk 3]
```diff
@@ -479,25 +479,16 @@ namespace PVR
...
      * @param channel The channel to open.
      * @return True if parental lock is overridden, false otherwise.
      */
     bool IsParentalLocked(const CPVRChannelPtr &channel);
 
     /*!
-     * @brief Open Numeric dialog to check for parental PIN.
-     * @param strTitle Override the title of the dialog if set.
-     * @return True if entered PIN was correct, false otherwise.
+     * @brief Restart the parantal timer.
```
Comment (Jalle19) [inline xbmc/pvr/PVRManager.h]: Typo

File: xbmc/pvr/PVRJobs.h [hunk 4]
```diff
@@ -49,4 +49,21 @@ namespace PVR
     bool DoWork() override;
   };
 
+  class CPVREventlogJob : public CJob
+  {
+  public:
+    CPVREventlogJob(bool bNotifyUser, bool bError, const std::string &label, const std::string &msg, const std::string &icon)
+    : m_bNotifyUser(bNotifyUser), m_bError(bError), m_label(label), m_msg(msg), m_icon(icon) {}
+    virtual ~CPVREventlogJob() {}
```
Comment (Jalle19) [inline xbmc/pvr/PVRJobs.h]: `= default()`?

File: xbmc/addons/PVRClient.cpp [hunk 5]
```diff
@@ -1618,77 +1592,6 @@ time_t CPVRClient::GetBufferTimeEnd(void) const
   return time;
 }
 
-bool CPVRClient::CanAutoconfigure(void) const
```
Comment (Jalle19) [inline xbmc/addons/PVRClient.cpp]: Are you sure all this Avahi auto-configuration stuff is unused?

**Comment by Jalle19** [comment]
Apart from the minors everything looks good. Hopefully this will decresae the chance of random deadlocks etc.

File: xbmc/addons/PVRClient.cpp (hunk 5 above)
Comment (ksooo) [inline xbmc/addons/PVRClient.cpp]: Yep, it is. I did not remove this feature but it is a matter of fact that this is dead code. AFAIK it never worked reliably, btw. Have seen many complaints in the forum in the past.

File: xbmc/pvr/PVRManager.h (hunk 3 above)
Comment (ksooo) [inline xbmc/pvr/PVRManager.h]: thanks

File: xbmc/pvr/PVRJobs.h (hunk 2 above)
Comment (ksooo) [inline xbmc/pvr/PVRJobs.h]: yep.

**Comment by ksooo** [comment]
jenkins build this please

**Comment by ksooo** [comment]
> library.mak:99: recipe for target 'libavcodec/avcodec-57.dll' failed
make: *** [libavcodec/avcodec-57.dll] Error 1

jenkins error is unrelated.

**Pull request merged**